### PR TITLE
feat(ai.training): handle cpu jobs

### PR DESCRIPTION
**Pull request by christophe-rannou**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         |no
| New feature?     | yes
| Breaking change? |no
| Tickets          | MANAGER-6761
| License          | BSD 3-Clause

## Description

This PR is related to the following jira task in AI Training project:

- MLS-1123: Handle CPU jobs in manager
- MLS-1068: Rework data step in job run form

### :flags: Translations

- [x] TRANS-34537

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/training/jobs/info/info.controller.js [hunk 1]
```diff
@@ -33,8 +33,17 @@ export default class PciTrainingJobsInfoController {
...
-    this.unitTax = this.getTax(1);
+    let resourceId;
+    let resourceN;
+    if (this.job.spec.resources.gpu >= 1) {
+      resourceId = 'gpu';
+      resourceN = this.job.spec.resources.gpu;
+    } else {
+      resourceId = 'cpu';
+      resourceN = this.job.spec.resources.cpu;
+    }
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/info/info.controller.js:44]:
```suggestion
    const resourceId = this.job.spec.resources.gpu >= 1 ? 'gpu': 'cpu';
    const resourceN = this.job.spec.resources.gpu ||  this.job.spec.resources.cpu
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html [hunk 2]
```diff
@@ -279,8 +279,10 @@
                                 ng-repeat="data in $ctrl.job.spec.volumes track by $index"
                                 class="api-list-item"
                             >
-                                {{ data.container }}@{{
-                                data.region}}:{{data.mountPath}}:{{data.permission}}
+                                {{ data.container }}@{{data.region}}<span
+                                    ng-if="data.prefix !== ''"
+                                    >/{{data.prefix}}</span
+                                >:{{data.mountPath}}:{{data.permission}}
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html:285]: Wouldn't it be easier to construct this string in a JS function ?

File: packages/manager/modules/pci/src/projects/project/training/jobs/jobs.routing.js [hunk 3]
```diff
@@ -7,11 +7,24 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          if (resourceId === 'gpu') {
+            catalogEntry =
+              pricesCatalog[`ai-training.ai1-1-gpu.minute.consumption`];
+          } else {
+            catalogEntry =
+              pricesCatalog[`ai-training.ai1-1-cpu.minute.consumption`];
+          }
+          return catalogEntry;
+        };
+      },
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/jobs.routing.js:22]:
```suggestion
      getCatalogEntryF: /* @ngInject */ (pricesCatalog) => (resourceId) => pricesCatalog[`ai-training.ai1-1-${resourceId}.minute.consumption`]
```
Also I'm not sure hardcoding plancode is the best idea for maintenance

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html [hunk 4]
```diff
@@ -130,6 +130,14 @@ <h2>
             data-editable="!$ctrl.isSubmit"
             data-on-submit="$ctrl.forceSubmitContainer(form)"
         >
+            <oui-message type="info" dismissable="true">
+                {{::
+                'pci_projects_project_training_jobs_submit_select_data_tips_1' |
+                translate}}<br />
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:136]: If the only purpose is to have a return to line, using an element with `display:block` should be done instead

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html [hunk 5]
```diff
@@ -167,6 +175,23 @@ <h2>
...
+                            data-label="{{:: 'pci_projects_project_training_jobs_submit_data_prefix' | translate  }}"
+                            data-label-popover="{{:: 'pci_projects_project_training_jobs_submit_data_prefix_help' | translate }}"
+                        >
+                            <input
+                                type="text"
+                                class="oui-input"
+                                placeholder=""
+                                data-ng-init=""
+                                name="prefix"
+                                data-ng-model="prefix"
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:189]:
```suggestion
                                data-ng-model="$ctrl.prefix"
```
If not `model` will be declared as part of the `$scope`

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html [hunk 6]
```diff
@@ -213,6 +238,18 @@ <h2>
...
                         </oui-field>
                     </oui-inline-adder-field>
+                    <oui-inline-adder-field data-adaptive>
+                        <oui-field
+                            data-label="{{:: 'pci_projects_project_training_jobs_submit_data_cache' | translate  }}"
+                            data-label-popover="{{:: 'pci_projects_project_training_jobs_submit_data_cache_help' | translate }}"
+                            size="auto"
+                        >
+                            <oui-checkbox
+                                data-ng-model="cache"
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:248]:
```suggestion
                                data-ng-model="$ctrl.cache"
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -35,9 +39,14 @@
   "pci_projects_project_training_jobs_submit_select_image_tips_help": "Les images hébergées sur une registry privé ou public sont acceptées (GitHub, GitLab, DockerHub, ...)",
   "pci_projects_project_training_jobs_submit_data_container": "Conteneur",
   "pci_projects_project_training_jobs_submit_data_mount_path": "Répertoire de montage",
+  "pci_projects_project_training_jobs_submit_data_prefix": "Filtrer par Prefix",
+  "pci_projects_project_training_jobs_submit_data_prefix_help": "Seuls les objets correspondants au prefix sont chargés dans le système de fichier",
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json:43]:
```suggestion
  "pci_projects_project_training_jobs_submit_data_prefix_help": "Seuls les objets correspondants au préfixe sont chargés dans le système de fichier",
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -35,9 +39,14 @@
   "pci_projects_project_training_jobs_submit_select_image_tips_help": "Les images hébergées sur une registry privé ou public sont acceptées (GitHub, GitLab, DockerHub, ...)",
   "pci_projects_project_training_jobs_submit_data_container": "Conteneur",
   "pci_projects_project_training_jobs_submit_data_mount_path": "Répertoire de montage",
+  "pci_projects_project_training_jobs_submit_data_prefix": "Filtrer par Prefix",
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_fr_FR.json:42]:
```suggestion
  "pci_projects_project_training_jobs_submit_data_prefix": "Filtrer par préfixe",
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html (hunk 2 above)
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html:285]: i think we can have something like that:

```
getVolumeText(volume) { 
   const prefix = volume.prefix ? volume.prefix : '';
   return `${volume.container}@${volume.region}${prefix}:${volume.mounthPath}`;
}
```

and use it:
```suggestion
      <span data-ng-bind="$ctrl.getVolumeText(volume)">/span>
```

File: packages/manager/modules/pci/src/projects/project/training/jobs/jobs.routing.js (hunk 3 above)
Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/jobs.routing.js:22]: It is clearly not, and that is something we need to rework

File: packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html (hunk 2 above)
Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html:285]: Clearly, doing based on @mohammed-zahaf suggestion

Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html:285]: Clearly, will do based on @mohammed-zahaf suggestion

Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/info/info.html:285]: Done, thanks for the suggestion

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html (hunk 4 above)
Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:136]: it is the only purpose, which element should I use ?

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html (hunk 5 above)
Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:189]: fixed

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html (hunk 6 above)
Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:248]: fixed

File: packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html (hunk 4 above)
Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html:136]:
```suggestion
                <span class="d-block" data-translate="pci_projects_project_training_jobs_submit_select_data_tips_1"></span>
```
or
```suggestion
                <p class="d-block" data-translate="pci_projects_project_training_jobs_submit_select_data_tips_1"></p>
```

Comment (christophe-rannou) [reply packages/manager/modules/pci/src/projects/project/training/jobs/submit/submit.html]: Fixed

**Comment by marie-j** [comment]
@christophe-rannou some changes don't seem related to this PR, maybe a rebase is needed ?

**Review by antleblanc — CHANGES_REQUESTED** [review]
Hi @christophe-rannou!

FYI #4523 has been merged into `develop` branch.
Could you please rebase your branch in order to apply a new review?

Thank you!

**Review by antleblanc — APPROVED** [review]

**Comment by antleblanc** [comment]
Pulling translations.

**Comment by antleblanc** [comment]
Translations has been pulled successfully.
There're now some conflicts.
I'm rebasing `feat-ai-training-cpu-job` branch on `develop` (got 1 approval).

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Review by antleblanc — APPROVED** [review]

**Review by marie-j — APPROVED** [review]

**Comment by marie-j** [comment]
@antleblanc Another rebase seems needed

**Comment by antleblanc** [comment]
> @antleblanc Another rebase seems needed

@marie-j  You're right!

This Pull Request requires #4831.
So once it will be merged, then I will rebase it on `develop` branch.

cc @christophe-rannou

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Review by antleblanc — APPROVED** [review]

**Comment by antleblanc** [comment]
Hi @christophe-rannou!

Following Pull Request #4561 has been recently merged on `develop` branch causing here some conflicts.
Could you please rebase and resolve existing conflicts?

Thank you!

**Review by antleblanc — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_de_DE.json [hunk 9]
```diff
@@ -43,7 +43,17 @@
...
+  "pci_projects_project_training_jobs_submit_select_data_tips_2": "Es hat sich bewährt, einen Container für Ihre Eingabedaten und einen zweiten für die Ausgabedaten hinzuzufügen.",
+  "pci_projects_project_training_jobs_submit_data_prefix": "Nach Präfix filtern",
+  "pci_projects_project_training_jobs_submit_data_prefix_help": "Nur Objekte mit dem entsprechenden Präfix werden in das Dateisystem geladen.",
+  "pci_projects_project_training_jobs_submit_data_permission_help": "Im „Nur-Lese-Modus“ wird am Ende des Auftrags keine Synchronisierung gestartet. Der Auftrag wird dadurch schneller abgeschlossen.",
+  "pci_projects_project_training_jobs_submit_data_cache": "Cache",
+  "pci_projects_project_training_jobs_submit_data_cache_help": "Container im Cache können ohne zusätzliche Synchronisierung durch andere Aufträge wiederverwendet werden. Wenn sie über einen bestimmten Zeitraum nicht verwendet werden, werden die Container aus dem Cache entfernt."
+  "pci_projects_project_training_jobs_resubmit_error": "Beim Neustart des Auftrags ist ein Fehler aufgetreten: {{ message }}",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/training/jobs/translations/Messages_de_DE.json:55]: Looks like the rebase introduced some invalid JSON files here.
I'm rebasing to address them.

**Review by antleblanc — APPROVED** [review]

**Comment by antleblanc** [comment]
Rebased successfully on `develop` branch.

**Pull request merged**